### PR TITLE
use python 3.9 to 3.11 multiple versions

### PR DESCRIPTION
**Pull request by jimexist**
expand GitHub action file to support multiple python version builds.

### CODE REVIEW

**Comment by goldmermaid** [comment]
Thanks for the PR and it looks like it has some compatibility issues with python 3.12 based on the log [here](https://github.com/CambioML/pykoi/actions/runs/6456750661/job/17718081026).

**Comment by Jimexist** [comment]
> Thanks for the PR and it looks like it has some compatibility issues with python 3.12 based on the log [here](https://github.com/CambioML/pykoi/actions/runs/6456750661/job/17718081026).

it looks like this requires an upgrade from [aiohttp][1]

[1]: https://github.com/aio-libs/aiohttp/issues/7646

File: .github/workflows/python-app.yml [hunk 1]
```diff
@@ -2,36 +2,39 @@ name: Python application
...
   contents: read
 
 jobs:
   build:
-
     runs-on: ubuntu-latest
 
+    strategy:
+      matrix:
+        python-version: ["3.9", "3.10", "3.11", "3.12"]
```
Comment (Jimexist) [inline .github/workflows/python-app.yml:18]:
```suggestion
        python-version: ["3.9", "3.10", "3.11"]
```

**Comment by Jimexist** [comment]
for the moment it makes sense to drop support for 3.12 as torch isn't supporting 3.12 yet.

**Pull request closed without merging**